src/components/editor/InputLang.jsx

```diff
@@ -15,47 +14,38 @@ class InputLang extends Component {
     }
   }
 
-  /*
-   * TODO:
-   * English is the default value, but is not set in the redux.lang.store, so it currently would need to be set manually
-   * in the generation of RDF.
-   * See https://github.com/LD4P/sinopia_editor/issues/290
-   */
-
-  /*
-   * When clicking Cancel make it not save the language. Clicking X to remove the input for the literal
-   * leaves the input in the lang redux store but nothing will be associated with it in the "literal" store.
-   * See https://github.com/LD4P/sinopia_editor/issues/275
-   */
-
   setPayLoad(items) {
     const payload = {
-      id: this.props.textValue,
+      id: this.props.textId,
+      reduxPath: this.props.reduxPath,
       items,
     }
 
-    this.props.handleSelectedChange(payload)
+    this.props.handleLangChange(payload)
   }
 
   createOptions = (json) => {
-    for (const i in json) {
-      try {
-        const item = Object.getOwnPropertyDescriptor(json, i)
-        const uri = item.value['@id']
-        const valArr = item.value['http://www.loc.gov/mads/rdf/v1#authoritativeLabel']
-        let label
-
-        valArr.forEach((obj) => {
-          if (obj['@language'] == 'en') {
-            label = obj['@value']
-          }
-        })
-
-        return { id: uri, uri, label }
-      } catch (error) {
-        // ignore
-      }
-    }
+    return json.reduce((result, item) => {
+      // Object.getOwnPropertyDescriptor is necessary to handle the @
+      const uri = Object.getOwnPropertyDescriptor(item, '@id').value
```

Comment:
Can also use the safe navigation operator here, which we use in a couple other spots in the editor:

```javascript
const uri = item?.['@id'].value
```

I am not sure which is more idiomatic or more readable.

Reply:
Or `item['@id']` if you don't need safe navigation.

Reply:
Agree with you that these are more idiomatic. However, for some unknown reason they don't work with `@id`.

Reply:
@justinlittman That's surprising! What kind of thing is `item` here? I created a dummy object in the console that contained a key of `@id` and it worked fine. So clearly you're not dealing with a dummy object here. :)